src/hooks/projects.ts

```diff
@@ -586,17 +586,34 @@ export function useRemoveMember({ projectId }: { projectId: string }) {
  * It is necessary to use this if you want the project role-related read hooks to update
  * based on role change events that are received in the background.
  *
+ * @param opts.callback Optional callback to invoke after cache invalidation when role changes
+ *
  * @example
  * ```tsx
  * function SomeComponent({ projectId }: { projectId: string }) {
  *   useProjectOwnRoleChangeListener({ projectId })
  * }
  * ```
+ *
+ * @example
+ * ```tsx
+ * function ComponentWithCallback({ projectId }: { projectId: string }) {
+ *   useProjectOwnRoleChangeListener({
+ *     projectId,
+ *     callback: () => {
+ *       // Handle role change, e.g., navigate to default project
+ *       console.log('Role changed!')
+ *     }
+ *   })
+ * }
+ * ```
  */
 export function useProjectOwnRoleChangeListener({
 	projectId,
+	callback,
 }: {
 	projectId: string
+	callback?: () => void
```

Comment:
can we rename this to `listener`? Think it's a more commonly used named parameter for these kinds of things.